deepchem/models/torch_models/gnn.py

```diff
@@ -0,0 +1,360 @@
+    >>> batched_graph = BatchGraphData(features).numpy_to_torch(device="cuda")
+    >>> modular = model = GNNModular("gin", 3, 64, 1, "attention", 0, "last", "edge_pred")
+    >>> gnnmodel = modular.gnn
+    >>> print(gnnmodel(batched_graph)[0].shape)
+    torch.Size([23, 64])
+
+    """
+
+    def __init__(self,
+                 atom_type_embedding,
+                 chirality_embedding,
+                 gconvs,
+                 batch_norms,
+                 dropout,
+                 JK,
+                 init_emb=False):
+        super(GNN, self).__init__()
+
+        self.atom_type_embedding = atom_type_embedding
+        self.chirality_embedding = chirality_embedding
+        self.gconv = gconvs
+        self.batch_norms = batch_norms
+        self.dropout = dropout
+        self.num_layer = len(gconvs)
+        self.JK = JK
+
+        # may mess with loading pretrained weights
+        if init_emb:
+            torch.nn.init.xavier_uniform_(self.atom_type_embedding.weight.data)
+            torch.nn.init.xavier_uniform_(self.chirality_embedding.weight.data)
+
+    def forward(self, data: BatchGraphData):
+        """
+        Forward pass for the GNN module.
+        """
+
+        x = self.atom_type_embedding(  # type: ignore
+            data.node_features[:, 0].long()) + self.chirality_embedding(
+                data.node_features[:, 1].long())
+
+        h_list = [x]
+        for i, conv_layer in enumerate(self.gconv):
+            h = conv_layer(h_list[i], data.edge_index, data.edge_features)
+            h = self.batch_norms[i](h)
+            h = F.dropout(F.relu(h), self.dropout, training=self.training)
+            if i == self.num_layer - 1:
+                # remove relu for the last layer
+                h = F.dropout(h, self.dropout, training=self.training)
+            else:
+                h = F.dropout(F.relu(h), self.dropout, training=self.training)
+            h_list.append(h)
+
+        # Different implementations of JK
+        if self.JK == "concat":
+            node_representation = torch.cat(h_list, dim=1)
+        elif self.JK == "last":
+            node_representation = h_list[-1]
+        elif self.JK == "max":
+            h_list = [h.unsqueeze_(0) for h in h_list]
+            node_representation = torch.max(torch.cat(h_list, dim=0), dim=0)[0]
+        elif self.JK == "sum":
+            h_list = [h.unsqueeze_(0) for h in h_list]
+            node_representation = torch.sum(torch.cat(h_list, dim=0), dim=0)[0]
+
+        return node_representation, data
+
+
+class GNN_head(torch.nn.Module):
+    """
+    Forward pass for the GNN head module.
+
+    Parameters
+    ----------
+    node_representation: torch.Tensor
+        The node representations after passing through the GNN layers.
+    data: BatchGraphData
+        The input graph data.
+
+    Returns
+    -------
+    out: torch.Tensor
+        The output of the GNN head module.
+    """
+
+    def __init__(self, pool, head):
```

Comment:
Can you document the parameters for the init? The parameters documented above are only for the forward pass

Reply:
Quick reminder here. The arguments documented above are for `forward` and not `__init__`

Reply:
Sorry got confused with the other PRs. This is fixed.